### ISSUE TITLE
Question about choice of microcontroller

### ISSUE THREAD

**Issue by msgilligan** [BlockchainCommons/bc-lethekit#61]
Why was the SparkFun Thing Plus - SAMD51 chosen? Is there any public documentation about the reasons for this choice.

Would it be possible to build the LetheKit with the $4 Raspberry Pi Pico? If not, why?

I'm asking for two reasons:
1. To better understand the current design decisions
2. I'm tempted to try to see if I can port this to the RP2040/Pico

**Comment by gorazdko** [comment]
> Why was the SparkFun Thing Plus - SAMD51 chosen? Is there any public documentation about the reasons for this choice?

I don't know about the reasons, @ksedgwic is the  original author who came up with the setup.

> Would it be possible to build the LetheKit with the $4 Raspberry Pi Pico? If not, why?

This could be doable if you stay within Arduino environment, otherwise I dont think it will be that easy.

Note also that lethekit uses cortex m4 whereas rpi uses m0+, so some compiling difficulties may arise from that. And wrt to resources (gpios, RAM, peripherals) I'd say the rpi should handle it.

**Comment by ksedgwic** [comment]
@msgilligan We were looking for an Arduino compatible device that didn't have radios (Bluetooth, WiFi, ..)

**Comment by ksedgwic** [comment]
I'm not a fan of the TRNG part anyway, I think all the entropy should be generated with dice.  Everything else is very portable ...

**Comment by msgilligan** [comment]
Thanks for the responses, @gorazdko and @ksedgwic!

The Pico doesn't have any radios, either. So it should be a good choice if it can be made to work. ($16 cheaper)

I've ordered the SAMD51 and the other major components and will try to build one. I'm experimenting with the Pico on another project. I may try to see if I can get the code to run on the Pico. I believe the Pico works with the Arduino environment, so it might not be that difficult.

**Comment by ksedgwic** [comment]
A big component of the price of the Thing Plus is the battery charging circuit.  If you can live w/o the battery things can be significantly cheaper.  Alternatively, if there was sufficient interest (quantity) it would make sense to design a PC board with the charging circuit provided externally.  A PC board would also simplify the assembly and make the result much more rugged ....

**Comment by msgilligan** [comment]
So with the Pico, you'd have to use a power adapter or a disposable battery pack?

**Comment by gorazdko** [comment]
> So with the Pico, you'd have to use a power adapter or a disposable battery pack

yes. You could also use rechargable batteries but you won't be able to recharge them via the board's usb

> There is also this board available

this one supports charging, but it's also more expensive
